lightstep/lightstep_b3_propagator.py

```diff
@@ -0,0 +1,127 @@
+from warnings import warn
+from re import compile as re_compile, IGNORECASE
+
+from opentracing.mocktracer.propagator import Propagator
+from opentracing.mocktracer.context import SpanContext
+from opentracing import span
+
+
+_SINGLE_HEADER = "b3"
+
+# Lower case is used here as the B3 specification recommends
+_TRACEID = "x-b3-traceid"
+_SPANID = "x-b3-spanid"
+_PARENTSPANID = "x-b3-parentspanid"
+_SAMPLED = "x-b3-sampled"
+_FLAGS = "x-b3-flags"
+
+
+class LightStepB3Propagator(Propagator):
+    """
+    Propagator for the B3 HTTP header format.
+
+    See: https://github.com/openzipkin/b3-propagation
+    """
+
+    def inject(self, span_context, carrier):
+        traceid = span_context.trace_id
+        if traceid is not None:
+            carrier[_TRACEID] = traceid
+        spanid = span_context.span_id
+        if spanid is not None:
+            carrier[_SPANID] = spanid
+        baggage = span_context._baggage
+        if baggage != span.SpanContext.EMPTY_BAGGAGE:
+            parentspanid = baggage.get(self._PARENTSPAN, None)
+            if parentspanid is not None:
+                carrier[_PARENTSPANID] = parentspanid
+            flags = baggage.get(self._FLAGS, None)
+            if flags is not None:
+                carrier[self._FLAGS] = flags
+            sampled = baggage.get(self._SAMPLED, None)
+            if sampled is not None:
+                if flags == 1:
+                    warn(
+                        "x-b3-flags: 1 implies x-b3-sampled: 1, not sending "
+                        "the value of x-b3-sampled"
+                    )
+                else:
+                    carrier[_SAMPLED] = sampled
+                    if sampled in [True, False]:
+                        warn(
+                            "The value of x-b3-sampled should "
+                            "be {} instead of {}".format(
+                                int(sampled), sampled
+                            )
+                        )
+        else:
+            if traceid is None or spanid is None:
```

Comment:
We don't want to raise an `Exception` here. And here we can totally use `warn` to signal the user about this.

Reply:
Ok, will downgrade to `warn`.

Reply:
Fixed too